src/ol/format/wkt.js

```diff
@@ -168,13 +169,25 @@ ol.format.WKT.encode_ = function(geom) {
   ol.DEBUG && console.assert(geometryEncoder, 'geometryEncoder should be defined');
   var enc = geometryEncoder(geom);
   type = type.toUpperCase();
+  if (geom instanceof ol.geom.SimpleGeometry) {
+    var layout = geom.getLayout();
+    var dimInfo = '';
+    if (layout === ol.geom.GeometryLayout.XYZ || layout === ol.geom.GeometryLayout.XYZM) {
+      dimInfo += 'Z';
+    }
+    if (layout === ol.geom.GeometryLayout.XYM || layout === ol.geom.GeometryLayout.XYZM) {
+      dimInfo += 'M';
+    }
+    if (dimInfo.length > 0) {
+      type += ' ' + dimInfo;
+    }
+  }
   if (enc.length === 0) {
     return type + ' ' + ol.format.WKT.EMPTY;
   }
   return type + '(' + enc + ')';
 };
 
-
```

Comment:
I'm not sure if we still enforce this automatically, but there need to be 2 lines here